src/Infrastructure/CreditCardExpiry.php

```diff
@@ -22,6 +22,15 @@ public function __construct( int $month, int $year ) {
 		$this->year = $year;
 	}
 
+	public static function newFromString( string $expirationDate ) {
+		$dateParts = explode( '/', $expirationDate );
+		if ( count( $dateParts ) === 2 ) {
+			return new self( (int)$dateParts[0], (int)$dateParts[1] );
+		}
+
+		return null;
```

Comment:
Do we really want to silently ignore invalid expiration dates and introduce null values into the system?

That question depends on context. Ignoring the context, I think it's nicer to throw exceptions in alternative constructors when they cannot construct an instance. Ofc the context question is more important.